crypto/SslHelper.h

```diff
@@ -32,7 +34,18 @@ class HMAC
     }
 
 private:
+#if OPENSSL_VERSION_MAJOR >= 3
+    EVP_MAC* _mac = nullptr;
+    EVP_MAC_CTX* _ctx;
+#else
     hmac_ctx_st* _ctx;
+#endif
+    std::unique_ptr<uint8_t[]> _key;
+    int32_t _keyLength;
```

Comment:
You could replace this 2 fields by a std::vector.
It is already a byte array allocated on heap memory and you its easier to use than deal with nullptr and size in a separated variable